docs/DeveloperGuide.adoc

```diff
@@ -297,59 +304,219 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+2. System updates the transaction repeater.
++
+Use case ends
+
+[discrete]
+=== Use case: Delete transaction repeaters including all produced transactions
+
+*MSS*
+
+1. User requests to delete a transaction repeater together with all its produced transactions.
+2. System removes existing transactions produced from transaction repeater.
+3. System removes transaction repeater.
++
+Use case ends
+
+[discrete]
+=== Use case: Filter transaction repeaters
+
+*MSS*
+
+1. User requests to find transaction repeaters by specified fields.
+2. System filters existing transaction repeaters against specified fields.
+3. System shows filtered transaction repeaters.
++
+Use case ends
+
+[discrete]
+=== Use case: Set monthly expenditure budget
+
+*MSS*
+
+1.  User requests to set the monthly expenditure budget.
+2.  System set the monthly expenditure budget.
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 1a. System detect an invalid budget amount.
++
+[none]
+** 1a1. System shows an error message.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+[discrete]
+=== Use case: View expenditure trend
+
+*MSS*
+
+1.  User requests to view expenditure trend over a period.
+2.  System retrieves expenditure entries for each month in the indicated period.
+3.  System totals up expenditure for each month in the indicated period.
+4.  System shows the expenditure totals for each month in the indicated period.
 +
-[none]
-** 3a1. AddressBook shows an error message.
+Use case ends.
+
+[discrete]
+=== Use case: View expenditure breakdown by category
+
+*MSS*
+
+1.  User requests to view expenditure breakdown by category over a period.
+2.  System retrieves expenditure entries for each category in the indicated period.
+3.  System totals up expenditure for each category in the indicated period.
+4.  System shows the expenditure totals for each category in the indicated period.
 +
-Use case resumes at step 2.
+Use case ends.
 
-_{More to be added}_
+[discrete]
+=== Use case: View expenditure ranking chart
+
+*MSS*
+
+1.  User requests to view expenditure ranking chart over a period.
+2.  System retrieves all expenditure entries in the indicated period.
+3.  System sorts expenditure entries in the indicated period.
+4.  System shows the sorted expenditure entries in the indicated period.
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 transactions without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
Should there be a well-defined benchmark to meet? (above average typing speed is subjective)